### DISCUSSION TITLE
[Unknown Mod][Locations][Hill Fort] Peasants present in the hero's army can be trained at the Hill Fort for a cost

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#3225] [Ideas]
The peasants present in the hero's army can be trained at the Hill Fort for a cost.
Peasants can be trained to Archers or Rangers or Pikemen or Veteran Pikemen, at the player choice, and the player choose the quantity of Peasants to be trained.

When the hero is positioned on the hill fort's active square, the "UPGRADE" button is displayed in the peasant information window in the hero's screen, allowing peasants to be upgraded at a cost.

The player has to pay a cost to upgrade the Peasants to the desired creatures:
- Peasants to Archers: 150 Gold per creature.
- Peasants to Rangers: 200 Gold per creature.
- Peasants to Pikemen: 200 Gold per creature.
- Peasants to Veteran Pikemen: 250 Gold per creature.

In fheroes2:

![Peasant upgrade 007](https://github.com/ihhub/fheroes2/assets/43583994/75daae20-3415-4dac-ad1e-999c6bcafa19)

![Peasant upgrade 011](https://github.com/ihhub/fheroes2/assets/43583994/079c7b30-f416-4120-8db6-4a56ff29d5a2)

![Peasant upgrade button 008](https://github.com/ihhub/fheroes2/assets/43583994/05ff8229-72a1-442b-8c0c-d058728d18f0)

In the yellow title of the window for upgrading peasants, the word "Recruit" is replaced by the words "Upgrade to" ("Upgrade to Rangers", etc.).
The left and right arrows on both sides of the creature sprite are displayed to allow to select the desired creature (Archers or Rangers or Pikemen or Veteran Pikemen).
![Peasant upgrade archer 011](https://github.com/ihhub/fheroes2/assets/43583994/de6c09f7-fb23-40fe-a202-a5f9571a898e)

![Peasant upgrade ranger 011](https://github.com/ihhub/fheroes2/assets/43583994/21feb442-4322-49b6-8691-6bc2e491198c)

![Peasant upgrade pikeman 011](https://github.com/ihhub/fheroes2/assets/43583994/e50256c7-d3d7-438c-9be2-222b6cefe0ec)

![Peasant upgrade veteran pikeman 011](https://github.com/ihhub/fheroes2/assets/43583994/d144bd6e-5569-4466-8378-f97e3c8f8f51)

This is an option:
"World: Peasants can be trained at the Hill Fort for a cost"

Images are for explanation only.

**Note: all of this could eventually change, especially the stated values.** :)

**Comment by lioror** [comment]
Is it available for download?

**Reply by LeHerosInconnu** [reply]
Hello @lioror,

For now, these are just ideas I'm putting in place before modding becomes available in fheroes2. :)

**Reply by lioror** [reply]
ohh is modding even planned?

**Reply by LeHerosInconnu** [reply]
Hello @lioror,

Yes: [Project roadmap](https://github.com/ihhub/fheroes2/wiki/Project-roadmap).
